api/test/integration/test_rbac_black_agents_endpoints.tavern.yaml

```diff
@@ -1319,6 +1319,53 @@ stages:
       json:
         <<: *permission_denied
 
+---
+test_name: PUT /groups/{group_id}/restart
+
+stages:
+
+    # PUT /groups/group1/restart
+  - name: Restart all agents from group1 (Denied)
+    request:
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/{version:s}/groups/group1/restart"
+      method: PUT
+      headers:
+        Authorization: "Bearer {test_login_token}"
+        content-type: "application/xml"
+    response:
+      status_code: 400
+      json:
+        title: "Wazuh Error"
+        detail: "Permission denied: Resource type: group:id"
+        status: 400
+        remediation: "Please, make sure you have permissions to execute current request, for more information on setting up permissions please visit XXXX"
```

Comment:
I would remove all checks except code 4000 to avoid excess compromise with the messages.